src/reducers/schemas/selectors.js

```diff
@@ -1,42 +1,140 @@
 import { createSelector } from 'reselect'
 
-// Groups schemas into groups
-// Ie. All Data Element related into 'dataElement'
-function schemaGroups(schemas) {
-    const groups = {}
-
-    schemas.forEach(s => {
-        if (!groups[s.group]) {
-            groups[s.group] = []
+/**
+ * Helpers
+ * =======
+ * =======
+ * =======
+ */
+
+/**
+ * Group and sort schemas
+ * ======================
+ */
+const groupSchemas = schemas =>
+    schemas.reduce((groups, schema) => {
+        const { group } = schema
+
+        return {
+            ...groups,
+            [group]: groups[group] ? [...groups[group], schema] : [schema],
         }
+    }, {})
+
+const combineSingleItemGroups = groups => {
+    const groupNames = Object.keys(groups)
 
-        groups[s.group].push(s)
-    })
+    return groupNames.reduce(
+        (combinedGroups, groupName) => {
+            const group = groups[groupName]
 
-    // check groups with 1 item and merge inside Other
-    const groupsWith1Item = []
-    const OTHER_GROUP_NAME = 'other'
-    let otherGroup = []
+            if (group.length === 1) {
+                return {
+                    ...combinedGroups,
+                    other: [...combinedGroups.other, group[0]],
+                }
+            }
 
-    Object.entries(groups).forEach(([k, v]) => {
-        if (v.length === 1) {
-            groupsWith1Item.push(k)
-            v[0]['group'] = OTHER_GROUP_NAME
-            otherGroup = otherGroup.concat(v)
+            return {
+                ...combinedGroups,
+                [groupName]: group,
+            }
+        },
+        { other: [] }
+    )
+}
+
+/**
+ * Get group label helpers
+ * =======================
+ */
+function breakOnCamelCase(schemaName, name) {
+    const temp = schemaName.substr(0, name.length).replace(/([A-Z]+)/g, ' $1')
+    return temp[0].toUpperCase() + temp.substr(1)
+}
```

Comment:
This could also use an explanatory comment

Reply:
Yeah I had my difficulties with these functions as well... I just copied them from the existing code base and now - after reading them again - I have no clue what's actually happening here.. 
I'll try to refactor this a little and add comment so it's clear what's happening

Reply:
I tried to improve this, can you check if it's more understandable now? @ismay